src/screens/HomeScreen.js

```diff
@@ -207,6 +229,23 @@ export const HomeScreen = ({ navigation, route }) => {
     globalSettings?.settings?.pushNotifications
   );
 
+  const { data: staticContentListData, refetch: staticContentListRefetch } = useStaticContent({
+    refreshTimeKey: `publicJsonFile-${staticContentName}`,
+    name: staticContentName,
+    type: 'json',
+    skip: !showStaticContentList
+  });
+
+  // function to add customised styles from `globalSettings` to `contentList`
+  const staticContentListItem = useMemo(
+    () =>
+      staticContentListData?.map((item: any) => ({
+        ...item,
+        appDesignSystem: appDesignSystem?.staticContentList
+      })),
```

Comment:
please check `appDesignSystem?.contentList` for presence before mapping through all `contentListData`. if there is no app design system, we do not need to loop over each item to return itself without an `appDesignSystem`

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/958/commits/3ca1913a818ab827bd028dbaa811645099164ea8